Makefile

```diff
@@ -0,0 +1,18 @@
+.PHONY: all
+all: build-bin build-docker deploy
+
+.PHONY: code-gen
+code-gen:
+	@./codegen/update-generated.sh
+
+.PHONY: build-bin
+build-bin: code-gen
+	@./build/build.sh
+
+.PHONY: build-docker
+build-docker: build-bin
+	@./build/docker_build.sh
+
+.PHONY: deploy
+deploy: build-docker
+	@./deploy/deployment.sh
```

Comment:
nit/suggestion: I usually like to have comments on each target and then a `make help` that prints each description, so it would be something like this: 

```
.PHONY: deploy
deploy: build-docker ## Builds docker image from go binary
	@./deploy/deployment.sh

.PHONY: help
help:
	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Reply:
Couldn't hurt, sure will add.